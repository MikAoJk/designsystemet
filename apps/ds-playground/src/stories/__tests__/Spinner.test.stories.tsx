import { JSX } from 'react';

import {
  Spinner,
  spinnerColorArr,
  spinnerPositionArr,
  spinnerSizeArr,
} from '@skatteetaten/ds-status';
import { expect } from '@storybook/jest';
import { Meta, StoryObj } from '@storybook/react';
import { within } from '@storybook/testing-library';

const meta = {
  component: Spinner,
  title: 'Tester/Spinner',
  argTypes: {
    // Baseprops
    key: { table: { disable: true } },
    ref: { table: { disable: true } },
    className: { table: { disable: true } },
    id: { table: { disable: true } },
    lang: { table: { disable: true } },
    'data-testid': { table: { disable: true } },
    // Props
    children: {
      table: { disable: true },
      control: 'text',
    },
    size: {
      table: { disable: true },
      control: 'inline-radio',
      options: [...spinnerSizeArr],
    },
    titlePosition: {
      table: { disable: true },
      control: 'inline-radio',
      options: [...spinnerPositionArr],
    },
    color: {
      table: { disable: true },
      control: 'inline-radio',
      options: [...spinnerColorArr],
    },
    hideTitle: {
      control: 'boolean',
      table: { disable: true },
    },
  },
} satisfies Meta<typeof Spinner>;
export default meta;
type Story = StoryObj<typeof meta>;

export const WithRef = {
  name: 'With Ref (FA1)',
  args: {
    ref: (instance: HTMLDivElement | null): void => {
      if (instance) {
        instance.id = 'dummyIdForwardedFromRef';
      }
    },
  },
  argTypes: {
    ref: { table: { disable: false } },
  },
  parameters: {
    imageSnapshot: { disable: true },
  },
  play: async ({ canvasElement }): Promise<void> => {
    const canvas = within(canvasElement);
    const spinner = canvas.getByRole('status');
    await expect(spinner).toHaveAttribute('id', 'dummyIdForwardedFromRef');
  },
} satisfies Story;

export const WithAttributes = {
  name: 'With Attributes (FA2-5)',
  args: {
    id: 'htmlid',
    className: 'dummyClassname',
    lang: 'nb',
    'data-testid': '123ID',
  },
  argTypes: {
    id: { table: { disable: false } },
    className: { table: { disable: false } },
    lang: { table: { disable: false } },
    'data-testid': { table: { disable: false } },
  },
  play: async ({ canvasElement }): Promise<void> => {
    const canvas = within(canvasElement);
    const spinner = canvas.getByRole('status');
    await expect(spinner).toHaveAttribute('id', 'htmlid');
    await expect(spinner).toHaveClass('dummyClassname');
    await expect(spinner).toHaveAttribute('lang', 'nb');
    await expect(spinner).toHaveAttribute('data-testid', '123ID');
  },
} satisfies Story;

export const Defaults = {
  name: 'Defaults (B2)',
  argTypes: {
    children: { table: { disable: false } },
  },
  play: async ({ canvasElement }): Promise<void> => {
    const canvas = within(canvasElement);
    const spinner = canvas.getByRole('status');
    await expect(spinner).toBeInTheDocument();
  },
} satisfies Story;

export const WithSizeAndPosition = {
  name: 'With Size, TitlePostition, Children (A1, A2, A4)',
  render: (args): JSX.Element => {
    return (
      <>
        <Spinner className={'bottomSpacingXL'} size={'small'} {...args} />
        <Spinner className={'bottomSpacingXL'} size={'medium'} {...args} />
        <Spinner className={'bottomSpacingXL'} size={'large'} {...args} />

        <Spinner
          className={'bottomSpacingXL'}
          titlePosition={'right'}
          size={'small'}
          {...args}
        />
        <Spinner
          className={'bottomSpacingXL'}
          titlePosition={'right'}
          size={'medium'}
          {...args}
        />
        <Spinner
          className={'bottomSpacingXL'}
          titlePosition={'right'}
          size={'large'}
          {...args}
        />
      </>
    );
  },
  args: {
    children: 'henter innhold',
  },
  argTypes: {
    children: { table: { disable: false } },
    size: { table: { disable: false } },
    titlePosition: { table: { disable: false } },
  },
} satisfies Story;

export const WithSizeAndPositionLongText = {
  ...WithSizeAndPosition,
  name: 'With Size, Text Postition, Children, Long Text (A1, A2, A4)',
  args: {
    children: 'Henter årsmiddelkurs fra valutakurstjenesten',
  },
};

export const WithColorsLightBackground = {
  name: 'With Color (A3)',
  render: (args): JSX.Element => {
    return (
      <>
        <Spinner
          className={'bottomSpacingXL'}
          color={'black'}
          size={'medium'}
          {...args}
        >
          {'Henter innhold'}
        </Spinner>
        <Spinner color={'blue'} size={'medium'} {...args}>
          {'Henter innhold'}
        </Spinner>
      </>
    );
  },
  argTypes: {
    color: {
      table: { disable: false },
    },
  },
} satisfies Story;

export const WithColorsDarkBackground = {
  name: 'With Color Dark Background(A3)',
  args: {
    color: 'white',
  },
  argTypes: {
    color: {
      table: { disable: false },
    },
  },
  parameters: {
    backgrounds: {
      default: 'dark',
    },
  },
} satisfies Story;

const spinnerText = 'laster data';
export const WithHideText = {
  name: 'With HideTitle (B1)',
  args: {
    hideTitle: true,
    children: spinnerText,
  },
  argTypes: {
    hideTitle: { table: { disable: false } },
  },
  play: async ({ canvasElement }): Promise<void> => {
    const canvas = within(canvasElement);
    // vi må bruke findBy i stedet for getBY her siden det er en liten delay før teksten rendres
    const spinnerTextElement = await canvas.findByText(spinnerText);
    await expect(spinnerTextElement).toBeInTheDocument();
  },
} satisfies Story;
