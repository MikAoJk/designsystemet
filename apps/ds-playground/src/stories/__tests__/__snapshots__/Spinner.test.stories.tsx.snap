// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tester/Spinner Defaults play-test 1`] = `
<div data-test-block="true">
  <div aria-live="polite"
       aria-atomic="true"
       class="Spinner_spinner__RJ9Al Spinner_spinner_black__-biVj Spinner_spinner_flexColumn__3SCpy Spinner_spinner_medium__kWbQU "
  >
    <div class="Spinner_spinnerAnimation__nAt8p ">
    </div>
    <span class="Spinner_spinnerTitle__4A7b4  Spinner_spinnerTitle_centerText__ud193">
      Laster inn
    </span>
  </div>
</div>
`;

exports[`Tester/Spinner WithAttributes play-test 1`] = `
<div data-test-block="true">
  <div aria-live="polite"
       aria-atomic="true"
       id="htmlid"
       class="Spinner_spinner__RJ9Al Spinner_spinner_black__-biVj Spinner_spinner_flexColumn__3SCpy Spinner_spinner_medium__kWbQU dummyClassname"
       lang="nb"
       data-testid="123ID"
  >
    <div class="Spinner_spinnerAnimation__nAt8p ">
    </div>
    <span class="Spinner_spinnerTitle__4A7b4  Spinner_spinnerTitle_centerText__ud193">
      Laster inn
    </span>
  </div>
</div>
`;

exports[`Tester/Spinner WithColorsDarkBackground smoke-test 1`] = `
<div data-test-block="true">
  <div aria-live="polite"
       aria-atomic="true"
       class="Spinner_spinner__RJ9Al Spinner_spinner_white__YLIF- Spinner_spinner_flexColumn__3SCpy Spinner_spinner_medium__kWbQU "
  >
    <div class="Spinner_spinnerAnimation__nAt8p ">
    </div>
    <span class="Spinner_spinnerTitle__4A7b4  Spinner_spinnerTitle_centerText__ud193">
      Laster inn
    </span>
  </div>
</div>
`;

exports[`Tester/Spinner WithColorsLightBackground smoke-test 1`] = `
<div data-test-block="true">
  <div aria-live="polite"
       aria-atomic="true"
       class="Spinner_spinner__RJ9Al Spinner_spinner_black__-biVj Spinner_spinner_flexColumn__3SCpy Spinner_spinner_medium__kWbQU bottomSpacingXL"
  >
    <div class="Spinner_spinnerAnimation__nAt8p ">
    </div>
    <span class="Spinner_spinnerTitle__4A7b4  Spinner_spinnerTitle_centerText__ud193">
      Henter innhold
    </span>
  </div>
  <div aria-live="polite"
       aria-atomic="true"
       class="Spinner_spinner__RJ9Al Spinner_spinner_blue__khcHU Spinner_spinner_flexColumn__3SCpy Spinner_spinner_medium__kWbQU "
  >
    <div class="Spinner_spinnerAnimation__nAt8p ">
    </div>
    <span class="Spinner_spinnerTitle__4A7b4  Spinner_spinnerTitle_centerText__ud193">
      Henter innhold
    </span>
  </div>
</div>
`;

exports[`Tester/Spinner WithHideText play-test 1`] = `
<div data-test-block="true">
  <div aria-live="polite"
       aria-atomic="true"
       class="Spinner_spinner__RJ9Al Spinner_spinner_black__-biVj Spinner_spinner_flexColumn__3SCpy Spinner_spinner_medium__kWbQU "
  >
    <div class="Spinner_spinnerAnimation__nAt8p ">
    </div>
    <span class="Spinner_spinnerTitle__4A7b4 Spinner_srOnly__eRxJZ Spinner_spinnerTitle_centerText__ud193">
      laster data
    </span>
  </div>
</div>
`;

exports[`Tester/Spinner WithRef play-test 1`] = `
<div data-test-block="true">
  <div aria-live="polite"
       aria-atomic="true"
       class="Spinner_spinner__RJ9Al Spinner_spinner_black__-biVj Spinner_spinner_flexColumn__3SCpy Spinner_spinner_medium__kWbQU "
       id="dummyIdForwardedFromRef"
  >
    <div class="Spinner_spinnerAnimation__nAt8p ">
    </div>
    <span class="Spinner_spinnerTitle__4A7b4  Spinner_spinnerTitle_centerText__ud193">
      Laster inn
    </span>
  </div>
</div>
`;

exports[`Tester/Spinner WithSizeAndPosition smoke-test 1`] = `
<div data-test-block="true">
  <div aria-live="polite"
       aria-atomic="true"
       class="Spinner_spinner__RJ9Al Spinner_spinner_black__-biVj Spinner_spinner_flexColumn__3SCpy Spinner_spinner_small__2mxjt bottomSpacingXL"
  >
    <div class="Spinner_spinnerAnimation__nAt8p ">
    </div>
    <span class="Spinner_spinnerTitle__4A7b4  Spinner_spinnerTitle_centerText__ud193">
      henter innhold
    </span>
  </div>
  <div aria-live="polite"
       aria-atomic="true"
       class="Spinner_spinner__RJ9Al Spinner_spinner_black__-biVj Spinner_spinner_flexColumn__3SCpy Spinner_spinner_medium__kWbQU bottomSpacingXL"
  >
    <div class="Spinner_spinnerAnimation__nAt8p ">
    </div>
    <span class="Spinner_spinnerTitle__4A7b4  Spinner_spinnerTitle_centerText__ud193">
      henter innhold
    </span>
  </div>
  <div aria-live="polite"
       aria-atomic="true"
       class="Spinner_spinner__RJ9Al Spinner_spinner_black__-biVj Spinner_spinner_flexColumn__3SCpy Spinner_spinner_large__QByjP bottomSpacingXL"
  >
    <div class="Spinner_spinnerAnimation__nAt8p ">
    </div>
    <span class="Spinner_spinnerTitle__4A7b4  Spinner_spinnerTitle_centerText__ud193">
      henter innhold
    </span>
  </div>
  <div aria-live="polite"
       aria-atomic="true"
       class="Spinner_spinner__RJ9Al Spinner_spinner_black__-biVj Spinner_spinner_flexRow__hdMnc Spinner_spinner_small__2mxjt bottomSpacingXL"
  >
    <div class="Spinner_spinnerAnimation__nAt8p ">
    </div>
    <span class="Spinner_spinnerTitle__4A7b4  ">
      henter innhold
    </span>
  </div>
  <div aria-live="polite"
       aria-atomic="true"
       class="Spinner_spinner__RJ9Al Spinner_spinner_black__-biVj Spinner_spinner_flexRow__hdMnc Spinner_spinner_medium__kWbQU bottomSpacingXL"
  >
    <div class="Spinner_spinnerAnimation__nAt8p ">
    </div>
    <span class="Spinner_spinnerTitle__4A7b4  ">
      henter innhold
    </span>
  </div>
  <div aria-live="polite"
       aria-atomic="true"
       class="Spinner_spinner__RJ9Al Spinner_spinner_black__-biVj Spinner_spinner_flexRow__hdMnc Spinner_spinner_large__QByjP bottomSpacingXL"
  >
    <div class="Spinner_spinnerAnimation__nAt8p ">
    </div>
    <span class="Spinner_spinnerTitle__4A7b4  ">
      henter innhold
    </span>
  </div>
</div>
`;

exports[`Tester/Spinner WithSizeAndPositionLongText smoke-test 1`] = `
<div data-test-block="true">
  <div aria-live="polite"
       aria-atomic="true"
       class="Spinner_spinner__RJ9Al Spinner_spinner_black__-biVj Spinner_spinner_flexColumn__3SCpy Spinner_spinner_small__2mxjt bottomSpacingXL"
  >
    <div class="Spinner_spinnerAnimation__nAt8p ">
    </div>
    <span class="Spinner_spinnerTitle__4A7b4  Spinner_spinnerTitle_centerText__ud193">
      Henter årsmiddelkurs fra valutakurstjenesten
    </span>
  </div>
  <div aria-live="polite"
       aria-atomic="true"
       class="Spinner_spinner__RJ9Al Spinner_spinner_black__-biVj Spinner_spinner_flexColumn__3SCpy Spinner_spinner_medium__kWbQU bottomSpacingXL"
  >
    <div class="Spinner_spinnerAnimation__nAt8p ">
    </div>
    <span class="Spinner_spinnerTitle__4A7b4  Spinner_spinnerTitle_centerText__ud193">
      Henter årsmiddelkurs fra valutakurstjenesten
    </span>
  </div>
  <div aria-live="polite"
       aria-atomic="true"
       class="Spinner_spinner__RJ9Al Spinner_spinner_black__-biVj Spinner_spinner_flexColumn__3SCpy Spinner_spinner_large__QByjP bottomSpacingXL"
  >
    <div class="Spinner_spinnerAnimation__nAt8p ">
    </div>
    <span class="Spinner_spinnerTitle__4A7b4  Spinner_spinnerTitle_centerText__ud193">
      Henter årsmiddelkurs fra valutakurstjenesten
    </span>
  </div>
  <div aria-live="polite"
       aria-atomic="true"
       class="Spinner_spinner__RJ9Al Spinner_spinner_black__-biVj Spinner_spinner_flexRow__hdMnc Spinner_spinner_small__2mxjt bottomSpacingXL"
  >
    <div class="Spinner_spinnerAnimation__nAt8p ">
    </div>
    <span class="Spinner_spinnerTitle__4A7b4  ">
      Henter årsmiddelkurs fra valutakurstjenesten
    </span>
  </div>
  <div aria-live="polite"
       aria-atomic="true"
       class="Spinner_spinner__RJ9Al Spinner_spinner_black__-biVj Spinner_spinner_flexRow__hdMnc Spinner_spinner_medium__kWbQU bottomSpacingXL"
  >
    <div class="Spinner_spinnerAnimation__nAt8p ">
    </div>
    <span class="Spinner_spinnerTitle__4A7b4  ">
      Henter årsmiddelkurs fra valutakurstjenesten
    </span>
  </div>
  <div aria-live="polite"
       aria-atomic="true"
       class="Spinner_spinner__RJ9Al Spinner_spinner_black__-biVj Spinner_spinner_flexRow__hdMnc Spinner_spinner_large__QByjP bottomSpacingXL"
  >
    <div class="Spinner_spinnerAnimation__nAt8p ">
    </div>
    <span class="Spinner_spinnerTitle__4A7b4  ">
      Henter årsmiddelkurs fra valutakurstjenesten
    </span>
  </div>
</div>
`;
